src/firmware/arm_firmware/armSketchRev3.ino

```diff
@@ -0,0 +1,763 @@
+   }
+}
+
+void a2AbsMove() {
+   a2.moveTo(step_in);
+   while(a2.distanceToGo() !=0 && !digitalRead(a2_sw)) {
+       a2.run();
+   }
+}
+
+void a3AbsMove() {
+   a3.moveTo(step_in);
+   while(a3.distanceToGo() !=0 && !digitalRead(a3_sw)) {
+       a3.run();
+   }
+}
+
+void a4AbsMove() {
+   a4.moveTo(step_in);
+   while(a4.distanceToGo() !=0 && !digitalRead(a4_sw)) {
+       a4.run();
+   }
+}
+
+void a5AbsMove() {
+   a5.moveTo(step_in);
+   while(a5.distanceToGo() !=0 && !digitalRead(a5_sw)) {
+       a5.run();
+   }
+}
+
+void a6AbsMove() {
+   a6.moveTo(step_in);
+   while(a6.distanceToGo() !=0 && !digitalRead(a6_sw)) {
+       a6.run();
+   }
+}
+
+// routine for if limit switches are triggered unexpectedly during a move
+void a1LimTrigger() {
+
+  if(digitalRead(a1_sw)) {
+     int axis = 1;
+     a1.setCurrentPosition(0);
+     a1.stop();
+     a1.move(homeComp[axis-1]);
+     a1.setMaxSpeed(a1_home_SPEED);
+     a1.setAcceleration(a1_home_ACC);
+        
+     while (a1.distanceToGo() != 0) {
+          a1.run();
+     }
+
+     a1.setCurrentPosition(0);
+     a1.setMaxSpeed(a1_SPEED);
+     a1.setAcceleration(a1_ACC);
+     angles[axis-1] = 0;
+  }
+}
+
+void a2LimTrigger() {
+
+  if(digitalRead(a2_sw)) {
+     int axis = 2;
+     a2.setCurrentPosition(0);
+     a2.stop();
+     a2.move(homeComp[axis-1]);
+     a2.setMaxSpeed(a2_home_SPEED);
+     a2.setAcceleration(a2_home_ACC);
+        
+     while (a2.distanceToGo() != 0) {
+          a2.run();
+     }
+
+     a2.setCurrentPosition(0);
+     a2.setMaxSpeed(a2_SPEED);
+     a2.setAcceleration(a2_ACC);
+     angles[axis-1] = 0;
+  }
+}
+
+void a3LimTrigger() {
+
+  if(digitalRead(a3_sw)) {
+     int axis = 3;
+     a3.setCurrentPosition(0);
+     a3.stop();
+     a3.move(homeComp[axis-1]);
+     a3.setMaxSpeed(a3_home_SPEED);
+     a3.setAcceleration(a3_home_ACC);
+        
+     while (a3.distanceToGo() != 0) {
+          a3.run();
+     }
+
+     a3.setCurrentPosition(0);
+     a3.setMaxSpeed(a3_SPEED);
+     a3.setAcceleration(a3_ACC);
+     angles[axis-1] = 0;
+  }
+}
+
+void a4LimTrigger() {
+
+  if(digitalRead(a4_sw)) {
+     int axis = 4;
+     a4.setCurrentPosition(0);
+     a4.stop();
+     a4.move(homeComp[axis-1]);
+     a4.setMaxSpeed(a4_home_SPEED);
+     a4.setAcceleration(a4_home_ACC);
+        
+     while (a4.distanceToGo() != 0) {
+          a4.run();
+     }
+
+     a4.setCurrentPosition(0);
+     a4.setMaxSpeed(a4_SPEED);
+     a4.setAcceleration(a4_ACC);
+     angles[axis-1] = 0;
+  }
+}
+
+void a5LimTrigger() {
+
+  if(digitalRead(a5_sw)) {
+     int axis = 5;
+     a5.setCurrentPosition(0);
+     a5.stop();
+     a5.move(homeComp[axis-1]);
+     a5.setMaxSpeed(a5_home_SPEED);
+     a5.setAcceleration(a5_home_ACC);
+        
+     while (a5.distanceToGo() != 0) {
+          a5.run();
+     }
+
+     a5.setCurrentPosition(0);
+     a5.setMaxSpeed(a5_SPEED);
+     a5.setAcceleration(a5_ACC);
+     angles[axis-1] = 0;
+  }
+}
+
+void a6LimTrigger() {
```

Comment:
This applies to the other functions above as well but I think the code could benefit from refactoring.

With the `LimTrigger` functions for instance, the only difference between `a5LimTrigger` and `a6LimTrigger` is that one operates on `a5` and the other operates on `a6`. It would be cleaner if there was a generic `LimTrigger` function that can accept `a5` or `a6` or whatever as a parameter, and then you wouldn't be repeating so much code.